packages/ember-application/lib/system/application.js

```diff
@@ -762,6 +776,10 @@ const Application = Engine.extend({
       _loaded.application = undefined;
     }
 
+    if (this._applicationInstances.length) {
+      this._applicationInstances.forEach(i => i.destroy());
```

Comment:
Should reset the array after looping through and destroying, thinking `this._applicaitonInstances.length = 0` or something...